go/pkg/pass1/set-path.go

```diff
@@ -86,199 +132,88 @@ func setpathObj(obj pathObj, intfToZone1 *routerIntf, distToZone1 int) (int, *lo
-			myLoop.clusterExit = cluster
-			return myLoop.clusterExit
-		}
+		return maxDistance, objLoop, partitionRouters
 	}
-}
 
-// Print list of names in messages.
-func nameList(namedZones []*zone) string {
-	var listOfZoneNames string
-	if len(namedZones) > 0 {
-		for _, myZone := range namedZones {
-			listOfZoneNames += " - "
-			listOfZoneNames += myZone.name
-			listOfZoneNames += "\n"
-		}
-	}
-	return listOfZoneNames
+	obj.setToZone1(intfToZone1)
+	return maxDistance, nil, partitionRouters
 }
 
-//##############################################################################
-// Purpose : Set direction and distances to an arbitrary chosen start zone.
-//           Identify loops inside the graph topology, tag nodes of a
-//           cycle with a common loop object and distance.
-//           Check for multiple unconnected parts of topology.
-func findDistsAndLoops() {
-	if len(zones) <= 0 { //Meike: warum <= ??
-		abort.Msg("topology seems to be empty")
-	}
+func extractPartitionsConnectedBySplitRouter(partitions []*zone,
+	partition2Routers map[*zone][]*router) []*zone {
 
-	var pathRouters []*router
-	for _, r := range allRouters {
-		if r.semiManaged {
-			pathRouters = append(pathRouters, r)
-		}
-		if r.managed != "" {
-			pathRouters = append(pathRouters, r)
-		}
-	}
-	startDistance := 0
-	var partitions []*zone
+	// Generate Lookups.
 	router2partition := make(map[*router]*zone)
 	partition2splitCrypto := make(map[*zone][]*router)
 
-	// Find one or more connected partitions in whole topology.
-	// Only iterate zones, because unconnected routers have been
-	// rejected before.
-	for _, z := range zones { //!!!Nicht wie den typ zone nennen!!!
-
-		// Zone is connected to some previously processed partition.
-		if z.getToZone1() != nil || z.getLoop() != nil {
-			continue
-		}
-
-		// Chose an arbitrary node to start from.
-		zone1 := z
-		//        debug zone1->{name};
-
-		// Traverse all nodes connected to zone1.
-		// Second parameter stands for not existing starting interface.
-		// Value must be "false" and unequal to any interface.
-		var max int
-		max, _ = setpathObj(zone1, zone1.toZone1, startDistance)
-
-		// Use other distance values in disconnected partition.
-		// Otherwise pathmark would erroneously find a path between
-		// disconnected objects.
-		startDistance = max + 1
-
-		// Collect zone1 of each partition.
-		partitions = append(partitions, zone1)
-
-		// Check if split crypto parts are located inside current partition.
-		// Collect remaining routers for next partititions.
-		var unconnected []*router
-		for _, r := range pathRouters {
-			if r.getToZone1() != nil || r.getLoop() != nil {
-				router2partition[r] = zone1
-				if r.origRouter != nil {
-					partition2splitCrypto[zone1] =
-						append(partition2splitCrypto[zone1], r)
-				}
-			} else {
-				unconnected = append(unconnected, r)
+	for partition := range partition2Routers {
+		for  _, r := range partition2Routers[partition] {
```

Comment:
Das geht schöner:
	for partition, routers := range partition2Routers {
		for _, r := range routers {